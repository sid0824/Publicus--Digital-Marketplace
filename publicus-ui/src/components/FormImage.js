import React from "react";

const defaultImage = "https://designshack.net/wp-content/uploads/placeholder-image.png";

// Component for rendering the image linked to a form
class FormImage extends React.Component {

  // Get the form image url from props
  getState = () => {
    const { image } = this.props;
    return {
      url: image || defaultImage
    }
  };

  state = this.getState();

  // Render default form image if an error occurs
  onError = () => {
    this.setState({
      url: defaultImage
    })
  };

  render() {
    const { url } = this.state;
    const { height, width } = this.props;
    return (
      <img
        id={"formImage"}
        alt={"formImage"}
        width={height || 270}
        height={width || 180}
        src={url}
        onError={this.onError}
      />
    )
  }

}

export default FormImage;