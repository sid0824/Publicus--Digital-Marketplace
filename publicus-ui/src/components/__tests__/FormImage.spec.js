import toJson from 'enzyme-to-json';
import React from "react";
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FormImage from "../FormImage";

configure({ adapter: new Adapter() });

describe('Form Image', () => {

  beforeEach(() => {

  });

  afterAll(() => {

  });

  test('should render form image', () => {
    const wrapper = shallow(
      <FormImage
        image={"https://mock-path.png"}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('should use default form image on error', () => {
    const wrapper = shallow(
      <FormImage
        image={"https://mock-path.png"}
      />
    );
    wrapper.instance().onError();
    expect(wrapper.state()).toMatchSnapshot();
  });


});