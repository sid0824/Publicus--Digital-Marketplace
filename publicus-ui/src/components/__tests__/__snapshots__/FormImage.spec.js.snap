// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Image should render form image 1`] = `
<img
  alt="formImage"
  height={180}
  id="formImage"
  onError={[Function]}
  src="https://mock-path.png"
  width={270}
/>
`;

exports[`Form Image should use default form image on error 1`] = `
Object {
  "url": "https://designshack.net/wp-content/uploads/placeholder-image.png",
}
`;
